### PR TITLE
fixed problems in #143

### LINKED ISSUE #143: `Valkey.set()` checks
hi

if we pass both `ex` and `px` to [set](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2248), it will be passed down to valkey, which thros a syntax error, but the error is not descriptive, so maybe adding a check would be good.

also there is no type check for [exat](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2307) and [pxat](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2312) and [keepttl](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2317) and [nx and xx](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2320-L2323)

also [get](https://github.com/valkey-io/valkey-py/blob/main/valkey/commands/core.py#L2325-L2327) looks like a bug, it should be the value of get that is passed down, not hardcoded `True`

the `options` dict also seems un-needed

let me know what you think

### PR DESCRIPTION
**Pull request by amirreza-sf80**
closes #143

### CODE REVIEW

**Comment by aiven-sal** [comment]
Thanks! Could you also add test cases?

**Comment by mkmkme** [comment]
Could you please add tests for both improvements you introduced?

**Comment by amirreza8002** [comment]
i pushed the last commit to be discussed here

i think since the type hint is explicit that no string should be passed, having those two lines is a bit weird

**Comment by mkmkme** [comment]
@aiven-sal we discussed with @amirreza8002 in matrix channel the change in [c4de8a5](https://github.com/valkey-io/valkey-py/pull/144/commits/c4de8a5d97ad3c0d080c4430f643cbca5164fc86). The current typing of the function does not allow passing a string as `ex`, so passing it at least will trigger the `mypy` warning. So this extra case looks really off, and it's quite inconsistent with the rest of the arguments. I agreed to remove it.

Is it okay with you?

**Comment by mkmkme** [comment]
@amirreza8002 the linting has failed. Could you fix it please?

**Comment by mkmkme** [comment]
So it seems that at least there were some users of `ex` being string :) So maybe we should not remove that handling...

**Comment by amirreza8002** [comment]
the test fail is for the last commit, I'll adjust the test if the change is approved,

sorry about the lint, I'm used to my ide doing it :/
I'll fix that up

**Comment by mkmkme** [comment]
So @amirreza8002 would you please revert the last commit instead of fixing the test?

We really don't want to break anything right now

**Comment by amirreza8002** [comment]
> My impression (which might be not 100% right) is that many valkey-py functions accept ints and numerical strings interchangeably. This may be not always intentional, but disallowing strings now is a breaking change. I tend to be against breaking changes, but if we do it in a major release I won't oppose it.

so do you think, for a major version, other parameters should alow strings, or should we remove the string support?
e.g. right now onle `ex` allows string, `px`, `exat` and `pxat` don't

**Comment by mkmkme** [comment]
>so do you think, for a major version, other parameters should alow strings, or should we remove the string support?
e.g. right now onle ex allows string, px, exat and pxat don't

I'd personally leave the arguments as they are currently. And we can put a mental note for the next major release to remove string support for `ex`

**Comment by mkmkme** [comment]
Hey @amirreza8002 ! Do I understand correctly that the changes are ready for review and to be merged?

If so, would you rebase your PR on the top of the current `main` branch?

**Comment by amirreza8002** [comment]
hi
yes, afaik it can be merged

**Comment by aiven-sal** [comment]
> hi is this a problem from my side? the test's error doesn't seem relevant to my code I'm not sure what the other fail is for

Hi.
The problem is that you merged instead of rebasing.
We don't want merge commits in PRs (they make the git history difficult to look back at).
To fix the problem you should rebase and remove the merge commit.

**Comment by mkmkme** [comment]
I think it should be fixed now, yes. I've approved the github actions.

>the problem was that i thought using github's sync would rebase, it seems like it doesn't 

AFAIK you can choose the way you want to update the branch.

File: valkey/commands/core.py [hunk 1]
```diff
@@ -2311,11 +2316,15 @@ def set(
             pieces.append("EXAT")
             if isinstance(exat, datetime.datetime):
                 exat = int(exat.timestamp())
+            elif not isinstance(exat, int):
```
Comment (aiven-sal) [inline valkey/commands/core.py:2319]: I'm not sure I understand this (and the other similar change below).
Shouldn't exat and pxat accept also strings? Or is there something else that prevents them from being string?

Comment (amirreza8002) [reply valkey/commands/core.py:2319]: I don't think it ever supported strings for exat,
am i missing something?
i can add support for strings, but my understanding of our conversation above was that i shouldn't, and the current string support for ex and px are to be removed in future

Comment (amirreza8002) [reply valkey/commands/core.py:2319]: i stand corrected, i'll remove the check

**Comment by amirreza8002** [comment]
i have some follow up for this, but i think it should be done in another pr

**Comment by mkmkme** [comment]
>i have some follow up for this, but i think it should be done in another pr

What do you have in mind?

**Comment by amirreza8002** [comment]
i have three things in mind
1. if valkey accepts strings, why cast it to `int` in `ex`

2. for px, exat and pxat it might be appropriate to add check to see if the string is digit, and not `hi`

3. if string is supported might as well add it to type hints

**Comment by mkmkme** [comment]
>if valkey accepts strings, why cast it to int in ex

I'm not a Valkey expert, but my understanding is that it only supports the string representations of numbers. But I'll have to double-check that.

In general, your questions are valid, but I'd hesitate to change any user experience at this moment.

**Review by mkmkme — APPROVED** [review]
The changes now look good to me.
Thanks! And sorry it was a bit painful to land :)

**Comment by amirreza8002** [comment]
> but I'd hesitate to change any user experience at this moment.

I'm not sure how this effects user xp, but i respect the packages development plans

I was just thinking ahead, i don't think I'll be working on it any time soon, probably do some type hints for the time being since I'm doing similar work in django-valkey

**Pull request merged**